azblob/url_container.go

```diff
@@ -181,7 +178,7 @@ func (p *AccessPolicyPermission) Parse(s string) error {
 func (c ContainerURL) SetAccessPolicy(ctx context.Context, accessType PublicAccessType, si []SignedIdentifier,
 	ac ContainerAccessConditions) (*ContainerSetAccessPolicyResponse, error) {
 	if ac.IfMatch != ETagNone || ac.IfNoneMatch != ETagNone {
-		panic("the IfMatch and IfNoneMatch access conditions must have their default values because they are ignored by the service")
+		sanityCheckFailed("the IfMatch and IfNoneMatch access conditions must have their default values because they are ignored by the service")
```

Comment:
This function already has an error on the return signature. Why not return an error here rather than a panic?

Reply:
The discussion will continue [here](https://github.com/Azure/azure-storage-blob-go/pull/73#discussion_r222069003).